src/main/kotlin/pw/forst/tools/katlib/IterableExtensions.kt

```diff
@@ -497,3 +497,52 @@ inline fun <TItem> Iterable<TItem>.itemsToString(
     }
     return sb.toString().restrictLengthWithEllipsis(totalLength)
 }
+
+/**
+ * Transforms an Iterable into nullable elements while only keeping the non-null elements.
+ */
+fun <T, R> Iterable<T>.flatMapIndexedNotNull(function: (index: Int, T) -> Iterable<R>?): List<R> {
+    return flatMapIndexedTo(ArrayList(), function)
+}
+
+private inline fun <T, R, C : MutableCollection<in R>> Iterable<T>.flatMapIndexedTo(destination: C,
+                                                                                    transform: (index: Int, T) -> Iterable<R>?): C {
+
```

Comment:
please format the file properly using IntelliJ (I intent to add some formatting/linting tool like detekt, but I haven't had time yet)
```
inline fun <T, R, C : MutableCollection<in R>> Iterable<T>.flatMapIndexedTo(
    destination: C,
    transform: (index: Int, T) -> Iterable<R>?
): C {
    forEachIndexed { index, element ->
        transform(index, element)?.let { elements ->
            destination.addAll(elements)
        }
    }
    
    return destination
}
```

Reply:
forgot to add code style schema for this fork, will correct it

Reply:
> understandings of properly

heh, true, `properly` is not correct wording 😄  -- I'm using the formatting, IntelliJ is using, so I just let IntelliJ format everything before I commit it

Reply:
I hit format on both of those files, and it seems to have resulted in a lot of changes.

Reply:
this formatting looks good to me 👍